src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/DSA/DSASignVerify.NistValidation.cs

```diff
@@ -0,0 +1,754 @@
+                "59d2cba33d6c1d9f3962330c1fcda7cfb18508fea7d0555e3a169daed353f3ee" +
+                "6f4bb30244319161dff6438a37ca793b24bbb1b1bc2194fc6e6ef60278157899" +
+                "cb03c5dd6fc91a836eb20a25c09945643d95f7bd50d206684d6ffc14d16d82d5" +
+                "f781225bff908392a5793b803f9b70b4dfcb394f9ed81c18e391a09eb3f93a03" +
+                "2d81ba670cabfd6f64aa5e3374cb7c2029f45200e4f0bfd820c8bd58dc5eeb34";
+
+            const string x =
+                "bd3006cf5d3ac04a8a5128140df6025d9942d78544e9b27efe28b2ca1f79e313";
+
+            const string y =
+                "00728e23e74bb82de0e1315d58164a5cecc8951d89e88da702f5b878020fd8d2" +
+                "a1791b3e8ab770e084ac2397d297971ca8708a30a4097d86740153ee2db6ab63" +
+                "43c5b6cc2c8a7fa59082a8d659931cc48a0433a033dbb2fff3aa545686f922c7" +
+                "063da1d52d9688142ec64a1002948e5da89165d9df8eed9aa469b61ee0210b40" +
+                "33562333097ba8659944e5f7924e04a21bc3edc6d551e202e4c543e97518f91e" +
+                "0cab49111029b29c3aa1bed5f35e5c90feb9d3c745953dbf859defce4537b4a0" +
+                "9801fdc8fe6999fbde39908079811b4b992c2e8333b9f800ea0d9f0a5f53607e" +
+                "308942e68efef01e03d7cca6f196872bf01f436d4a8e05fc59d8fbc6b88a166f" +
+                "57a4e99d67ddaece844653be77819747dd2e07d581c518cb9779e9f7960c17ff" +
+                "0bae710ecf575b09591b013b4805c88b235df262e61a4c94f46bf9a08284611d" +
+                "f44eadd94f44cef6225a808e211e4d3af5e96bce64a90f8013874f10749a8382" +
+                "a6026a855d90853440bfce31f258b3a258f7b5e659b43e702dee7c24c02d2284";
+
+            const string msg =
+                "baeb12a1ebd8057a99a0137ee60f60eed10d26f1eab22ae2d9adbc3e5ffc3252" +
+                "abf62b614707ad2546141bed779f0cfad9544a74e562da549e2f7b286efb6154" +
+                "49b0946dc7c498d8f12150b2eacbd27157966f592ad5f3e43a24c60b7e06630b" +
+                "82a4fdb699119dbd878b13a98bf22a7b3dc7efdd992ce6b8a950e61299c5663b";
+
+            const string r =
+                "8d357b0b956fb90e8e0b9ff284cedc88a04d171a90c5997d8ee1e9bc4d0b35ff";
+
+            const string s =
+                "ab37329c50145d146505015704fdc4fb0fd7207e0b11d8becbad934e6255c30c";
+
+            Validate(p, q, g, x, y, msg, r, s, HashAlgorithmName.SHA512);
+        }
+
+        private static void Validate(
+            string p,
+            string q,
+            string g,
+            string x,
+            string y,
+            string msg,
+            string r,
+            string s,
+            HashAlgorithmName hashAlgorithm)
+        {
+            // Public+Private key
+            using (DSA dsa = DSAFactory.Create())
+            {
+                dsa.ImportParameters(
+                    new DSAParameters
+                    {
+                        P = p.HexToByteArray(),
+                        Q = q.HexToByteArray(),
+                        G = g.HexToByteArray(),
+                        X = x.HexToByteArray(),
```

Comment:
Curious as why test VerifyData with and without private key; has there ever an issue with specifying a private key for verify scenarios?